dashboards_bundlers/server_upload/__init__.py

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import os
+import requests
+from notebook.utils import url_path_join
+from tornado import escape
+from tornado.log import access_log
+from ..local_deploy import bundle_file_references
+
+ENDPOINT = '/notebooks/'
+
+def bundle(handler, abs_nb_path):
+    '''
+    Uploads a notebook to a Jupyter Dashboard Server
+    '''
+    # Get name of notebook from filename
+    notebook_basename = os.path.basename(abs_nb_path)
+    notebook_name = os.path.splitext(notebook_basename)[0]
+    if 'DASHBOARD_SERVER_URL' in os.environ:
+        server_url = url_path_join(os.getenv('DASHBOARD_SERVER_URL'), ENDPOINT, escape.url_escape(notebook_name, False))
+        with open(abs_nb_path, 'rb') as notebook:
+            headers = {}
+            if 'DASHBOARD_SERVER_TOKEN' in os.environ:
+                headers = headers + {'Authorization':'token ' + os.getenv('DASHBOARD_SERVER_TOKEN')}
+            result = requests.post(server_url, files={'file': notebook}, headers=headers)
+            result.raise_for_status()
+    else:
+        access_log.debug('Can not deploy, DASHBOARD_SERVER_URL not set')
```

Comment:
Bundler needs to finish the web response at the end. On success, response should be a redirect to the dashboard server URL with the `/notebooks/<name of notebook>` appended. (Again, this is going to change shortly in jupyter-incubator/dashboards_nodejs_app#48. When the DASHBOARD_SERVER_URL is not set, `handler.send_error` or raise a Tornado web error with status 500 and a custom message which the Jupyter error page will show in the page. When any other error occurs, let it propagate (like you're doing).